File: Tutorial/Examples/Competing consumers/Producer.py
import pika
import time
import random

# Set connection parameters. If connecting to a real server localhost should be replaced by the server´s address.
connection_parameters = pika.ConnectionParameters('localhost')

# Instantiate a connection unsing the connection_parameters previously defined.
connection = pika.BlockingConnection(connection_parameters)

# Instantiate a channel.
channel = connection.channel()

# Instantiate a queue.
channel.queue_declare(queue='CCQueue')

# Define an id for the message to send.
messageId = 1

# Continuously send messages whit a random delay between them.
while (True):

    # Compose the message body.
    message = f"This is message No. {messageId}"

    # Publish the message.
    channel.basic_publish(exchange='', routing_key='CCQueue', body=message)

    print(f"sent message: {message}")

    # Calculate delay.
    time.sleep(random.randint(1, 4))

    # Update the id for the next message.
    messageId += 1

File: Tutorial/Examples/Competing_consumers/Producer_C#/Producer.cs
﻿using System;
using System.Collections.Generic; 
using System.Threading.Tasks;
using System.Linq;
using System.Text;
using RabbitMQ.Client;

class Program{
    static void Main(string[] args){
        //Variable declaration.
        int h;

        // Create connection. When connecting to a real server, localhost should be replaced by the server´s address.
        var factory = new ConnectionFactory{ HostName = "localhost"};

        // Instantiate the connection and one channel.
        using var connection = factory.CreateConnection();
        using var channel = connection.CreateModel();

        // Declare a queue called CCQueue.
        channel.QueueDeclare(
            queue : "CCQueue", 
            durable: false, 
            exclusive: false, 
            autoDelete: false,
            arguments: null);

        // Variable setup for message construction.
        int count = 0;
        int adder = 1;
        int[] A ={0,0,0};
        int[] B ={1,0,0};
        int[] vector;

        try{
            h = int.Parse(args[0]);
        }
        catch{
            h = 20;
            Console.WriteLine("Invalid argument. Default value asigned");

        }

        while (true)
        {   
            // Compose the message's body.
            if (count % 2 == 0){
                vector = A.Take(3).ToArray();
            }
            else{
                vector = B.Take(3).ToArray();
            }
            vector[1] = count;

            // Declare message to produce and encode it as bytes.
            string message = string.Join(",", vector);
            var body = Encoding.UTF8.GetBytes(message);

            // Start publishing to CCQueue.
            channel.BasicPublish("", "CCQueue", null, body);

            // Print the published message to console.
            Console.WriteLine($"Send message: {message}");

            // Add a delay that simulates processing time.
            Task.Delay(TimeSpan.FromSeconds(0.1)).Wait();

            // Update the id for the next message.
            if (count > h || count < 0){
                adder *= -1;
            }
            count += adder;
        }
    }
}